crates/re_log_types/src/data_table_batcher.rs

```diff
@@ -0,0 +1,787 @@
+        self.send_cmd(flush_cmd);
+        oneshot.recv().ok();
+    }
+
+    fn send_cmd(&self, cmd: Command) {
+        // NOTE: Internal channels can never be closed outside of the `Drop` impl, this cannot
+        // fail.
+        self.tx_cmds.send(cmd).ok();
+    }
+}
+
+#[allow(clippy::needless_pass_by_value)]
+fn batching_thread(
+    config: DataTableBatcherConfig,
+    rx_cmd: Receiver<Command>,
+    tx_table: Sender<DataTable>,
+) {
+    let rx_tick = crossbeam::channel::tick(config.flush_tick);
+
+    struct Accumulator {
+        latest: Instant,
+        pending_rows: Vec<DataRow>,
+        pending_num_rows: u64,
+        pending_num_bytes: u64,
+    }
+
+    impl Accumulator {
+        fn reset(&mut self) {
+            self.latest = Instant::now();
+            self.pending_rows.clear();
+            self.pending_num_rows = 0;
+            self.pending_num_bytes = 0;
+        }
+    }
+
+    let mut acc = Accumulator {
+        latest: Instant::now(),
+        pending_rows: Default::default(),
+        pending_num_rows: Default::default(),
+        pending_num_bytes: Default::default(),
+    };
+
+    fn do_push_row(
+        config: &DataTableBatcherConfig,
+        acc: &mut Accumulator,
+        mut row: DataRow,
+    ) -> bool {
+        // TODO(#1760): now that we're re doing this here, it really is a massive waste not to send
+        // it over the wire...
+        row.compute_all_size_bytes();
+
+        acc.pending_num_rows += 1;
+        acc.pending_num_bytes += row.total_size_bytes();
+        acc.pending_rows.push(row);
+
+        acc.pending_num_rows >= config.flush_num_rows
+            || acc.pending_num_bytes >= config.flush_num_bytes
+    }
+
+    fn do_flush_all(acc: &mut Accumulator, tx_table: &Sender<DataTable>, reason: &str) {
+        let rows = &mut acc.pending_rows;
+
+        if rows.is_empty() {
+            return;
+        }
+
+        re_log::trace!(reason, "flushing tables");
+
+        let table = DataTable::from_rows(TableId::random(), rows.drain(..));
+        // TODO(#1981): efficient table sorting here, following the same rules as the store's.
+        // table.sort();
+
+        // NOTE: This can only fail if all receivers have been dropped, which simply cannot happen
+        // as long the batching thread is alive... which is where we currently are.
+        tx_table.send(table).ok();
+    }
+
+    use crossbeam::select;
+    loop {
+        select! {
+            recv(rx_cmd) -> cmd => {
+                let Ok(cmd) = cmd else {
+                    // All command senders are gone, which can only happen if the
+                    // `DataTableBatcher` itself has been dropped.
+                    break;
+                };
+
+                match cmd {
+                    Command::AppendRow(row) => {
+                        if do_push_row(&config, &mut acc, row) {
+                            do_flush_all(&mut acc, &tx_table, "bytes|rows");
+                            acc.reset();
```

Comment:
We call `acc.reset();` every time we call `do_flush_all()` which has access to `mut acc`. Why not reset at the end of `do_flush_all()`?